sstables/compaction.hh

```diff
@@ -84,6 +90,18 @@ namespace sstables {
         compaction_type type() const;
     };
 
+    struct compaction_completion_desc {
+        std::vector<shared_sstable> input_sstables;
+        std::vector<shared_sstable> output_sstables;
+        // Set of compacted partition ranges that should be invalidated in the cache.
+        dht::partition_range_vector ranges_for_cache_invalidation;
+    };
+
+    // creates a new SSTable for a given shard
+    using creator_fn = std::function<shared_sstable(unsigned shard)>;
```

Comment:
why not `shard_id shard` as it used to be?

Reply:
oversight.